xbmc/cores/VideoPlayer/Process/VideoBuffer.h

```diff
@@ -99,7 +99,7 @@ class CVideoBuffer
   virtual uint8_t* GetMemPtr() { return nullptr; };
   virtual void GetPlanes(uint8_t*(&planes)[YuvImage::MAX_PLANES]) {};
   virtual void GetStrides(int(&strides)[YuvImage::MAX_PLANES]) {};
-  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES]) {};
+  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES], const int planeOffsets[YuvImage::MAX_PLANES]) {};
```

Comment:
Its not as ref because is optional. You can leave it nullptr and have the previous plane offset calculation. Sure I could make a new call with 4, but this would lead to not small code duplication in CVideoBufferSysMem.